src/Domain/BlockLogicalCoordinates.cpp

```diff
@@ -63,11 +63,33 @@ std::vector<block_logical_coord_holder<Dim>> block_logical_coordinates(
         } else if constexpr (std::is_same_v<Frame, ::Frame::Distorted>) {
           // Point is in the distorted frame, so we need to map to the grid
           // frame and then the logical frame.
+          if (not block.has_distorted_frame()) {
+            // Note that block.has_distorted_frame() can be different for
+            // different Blocks.  However, the template parameter Frame is
+            // compile-time and is the same for all Blocks.
+            //
+            // Explanation of the logic here:
+            // 1. Recall that block_logical_coordinates loops through all the
+            //    Blocks, and skips all the Blocks except for the first Block
+            //    it finds that contains the point x.
+            // 2. If Frame is ::Frame::Distorted but
+            //    block.has_distorted_frame() is false, then this block
+            //    cannot contain the point x. Therefore, we should simply
+            //    skip this block.  If it turns out that no blocks contain
+            //    the point x, then we will get an error later.
+            //    (Note that our primary use case for ::Frame::Distorted is to
+            //    find an apparent horizon in the distorted frame. In that
+            //    case, only the Blocks near a horizon have a distorted frame
+            //    because only those Blocks have distortion maps. Thus,
+            //    the Blocks that are skipped here are those that are far
+            //    from horizons).
```

Comment:
This means that you can't use `block_logical_coordinates` in its current form to resolve `Distorted` -> `Logical` over the full grid. Calling `block_logical_coordinates` with distorted-frame coordinates will only return blocks that have a distorted frame. That's probably fine, but please document this clearly in the dox for this function.